packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -644,18 +645,30 @@ class DragTarget<T extends Object> extends StatefulWidget {
   _DragTargetState<T> createState() => _DragTargetState<T>();
 }
 
-List<T?> _mapAvatarsToData<T extends Object>(List<_DragAvatar<T>> avatars) {
-  return avatars.map<T?>((_DragAvatar<T> avatar) => avatar.data).toList();
+List<T?> _mapAvatarsToData<T extends Object>(List<_DragAvatar<Object>> avatars) {
+  return avatars.map<T?>((_DragAvatar<Object> avatar) => avatar.data as T?).toList();
 }
 
 class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
-  final List<_DragAvatar<T>> _candidateAvatars = <_DragAvatar<T>>[];
+  final List<_DragAvatar<Object>> _candidateAvatars = <_DragAvatar<Object>>[];
   final List<_DragAvatar<Object>> _rejectedAvatars = <_DragAvatar<Object>>[];
 
+  bool _checkDataType(Object? data, Type type) {
+    if (!kIsWeb) {
+      return data is T?;
+    } else {
+      if ((type == int && T == double) || (type == double && T == int)) {
```

Comment:
We must use the `type` that passed in, can not use `data.runtimeType` because `int` and `double` are `int` runtime type on WEB.